src/eredis_client.erl

```diff
@@ -559,21 +568,21 @@ maybe_reconnect(Reason,
 %% `{connection_ready, Socket}'.
 reconnect_loop(Client, ReconnectSleep, Host, Port, SocketOptions,
                TlsOptions, ConnectTimeout, AuthCmd, Db) ->
-    receive
-        {'EXIT', Client, Reason} -> exit(Reason)
-    after
-        ReconnectSleep ->
-            Client ! reconnect_attempt,
-            case connect(Host, Port, SocketOptions, TlsOptions, ConnectTimeout,
```

Comment:
Here, I just moved the `receive...after ReconnectSleep` block to just before the recursive call. It was at the beginning of the function. There are no other changes to this function.

Reply:
Nice solution.